src/components/core/Button/Button.tsx

```diff
@@ -1,107 +1,118 @@
-import { memo, useMemo } from 'react';
-import styled from 'styled-components';
+import styled, { css } from 'styled-components';
 import Loader from '../Loader/Loader';
-import { TitleXS } from '../Text/Text';
 import colors from '../colors';
 import transitions from '../transitions';
+import { AnchorHTMLAttributes, ButtonHTMLAttributes } from 'react';
+import { BODY_FONT_FAMILY } from '../Text/Text';
+import Link from 'next/link';
 
-type ButtonStyle = 'primary' | 'secondary';
-
-type Props = {
-  type?: ButtonStyle;
-  className?: string;
-  text?: string;
-  mini?: boolean;
-  onClick?: () => void;
-  disabled?: boolean;
-  loading?: boolean;
-  dataTestId?: string;
-};
-
-function Button({
-  type = 'primary',
-  className,
-  text,
-  mini = false,
-  onClick,
-  disabled,
-  loading,
-  dataTestId,
-}: Props) {
-  const ButtonComponent = useMemo(
-    () => (type === 'primary' ? StyledPrimaryButton : StyledSecondaryButton),
-    [type]
-  );
-  return (
-    <ButtonComponent
-      // Renaming this prop `buttonStyle` since the `type` prop
-      // already exists for styled components
-      buttonStyle={type}
-      className={className}
-      onClick={onClick}
-      disabled={disabled}
-      data-testid={dataTestId}
-      mini={mini}
-    >
-      {loading ? (
-        <Loader inverted size={mini ? 'mini' : 'small'} />
-      ) : (
-        <TitleXS color={type === 'primary' ? colors.white : colors.shadow}>{text}</TitleXS>
-      )}
-    </ButtonComponent>
-  );
-}
+// TODO:
+// - should a `loading` button be disabled/non-interactive?
+// - should `Loader` be conditionally inverted based on `variant`?
+// - should `Loader` just always be `mini`?
+// - why is opacity on disabled+loading so much stronger than just disabled?
```

Comment:
These are questions I had as I refactored the `Button` while trying to maintain backwards compatibility.